sklearn/_loss/_loss.pyx.tp

```diff
@@ -818,6 +818,9 @@ cdef inline double_pair cgrad_hess_exponential(
 cdef class CyLossFunction:
     """Base class for convex loss functions."""
 
+    def __reduce__(self):
```

Comment:
Is it needed actually? It seems to me like `CyLossFunction` is only used as a base class and you are not not really meant to create an instance of `CyLossFunction`.

Reply:
this is needed since not all losses have a parameter, which means they won't have our custom `__reduce__` generated for them. Note the `{{if param is not None}}` around the `__reduce__` implementation bellow.